python/cudf/cudf/core/series.py

```diff
@@ -2572,6 +2578,22 @@ def repeat(self, repeats, axis=None):
         new_index = self.index.repeat(repeats)
         return Series(data, index=new_index, name=self.name)
 
+    def _align_to_index(self, index, how="outer", allow_non_unique=False):
+        """
+        Align to the given Index. See _align_indices below.
+        """
+        index = as_index(index)
+        if self.index.equals(index):
+            return self
+        if not allow_non_unique:
+            if len(self) != len(self.index.unique()) or len(index) != len(
+                index.unique()
+            ):
+                raise ValueError("Cannot align indices with non-unique values")
+        lhs = self.to_frame(0)
+        rhs = cudf.DataFrame(index=as_index(index))
+        return lhs.join(rhs, how=how, sort=True).iloc[:, 0].loc[index]
```

Comment:
is calling the `loc` here just to get the ordering correct based on `index`?

Reply:
Yep. Maybe we can save some effort by setting `sort=False`?